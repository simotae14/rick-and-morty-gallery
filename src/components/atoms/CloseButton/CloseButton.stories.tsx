import type { Meta, StoryObj } from '@storybook/react';

import CloseButton from './index';

const meta = {
  title: 'Atoms/CloseButton',
  component: CloseButton,
  tags: ['autodocs'],
  argTypes: {
    onClickClose: {
      action: 'clicked',
    },
  },
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof CloseButton>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
