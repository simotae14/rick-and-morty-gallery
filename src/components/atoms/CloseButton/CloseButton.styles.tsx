import styled from 'styled-components';

export const CloseButtonContainer = styled.div`
  width: 32px;
  height: 32px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(18, 28, 51, 0.1);
  border-radius: 8px;
  cursor: pointer;
  position: relative;
`;

export const CloseIcon = styled.span`
  position: absolute;
  right: 0;
  top: 0;
  width: 32px;
  height: 32px;
  opacity: 0.7;
  &hover {
    opacity: 1;
  }
  &::after {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: #333;
  }
  &::before {
    position: absolute;
    left: 15px;
    content: ' ';
    height: 33px;
    width: 2px;
    background-color: #333;
  }
  &::before {
    transform: rotate(45deg);
  }
  &::after {
    transform: rotate(-45deg);
  }
`;
