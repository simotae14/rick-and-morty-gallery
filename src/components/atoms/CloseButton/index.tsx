import { CloseIcon, CloseButtonContainer } from './CloseButton.styles';

interface CloseIconProps {
  /**
   * Is the handler to close the button
   */
  onClickClose: () => void;
}

/**
 * Component used for the 404 page and on loading
 */
function CloseButton({ onClickClose }: CloseIconProps) {
  return (
    <CloseButtonContainer onClick={onClickClose}>
      <CloseIcon />
    </CloseButtonContainer>
  );
}

export default CloseButton;
