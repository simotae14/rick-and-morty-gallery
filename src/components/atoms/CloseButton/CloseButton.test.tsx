/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-extraneous-dependencies */
import { describe, it, vi } from 'vitest';
// eslint-disable-next-line import/no-extraneous-dependencies
import { render, fireEvent } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';

import CloseButton from '.';

expect.extend(toHaveNoViolations);

const mockedProps = {
  onClickClose: vi.fn(),
};

describe('Components => atoms => CloseButton', () => {
  afterEach(() => {
    vi.clearAllMocks();
    vi.resetAllMocks();
  });
  it('should render the CloseButton even with click handler', () => {
    const { asFragment } = render(<CloseButton {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should click the Close Button and trigger the click handler', () => {
    const { container } = render(<CloseButton {...mockedProps} />);
    const closeBtn = container.querySelector('div');
    if (closeBtn) {
      fireEvent.click(closeBtn);
      expect(mockedProps.onClickClose).toHaveBeenCalledTimes(1);
    }
  });

  it('should not fail any accessibility tests', async () => {
    const { container } = render(<CloseButton {...mockedProps} />);
    expect(await axe(container)).toHaveNoViolations();
  });
});
