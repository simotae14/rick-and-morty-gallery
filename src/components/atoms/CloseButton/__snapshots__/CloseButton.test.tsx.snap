// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components => atoms => CloseButton > should render the CloseButton even with click handler 1`] = `
<DocumentFragment>
  <div
    class="sc-eLqIir godZbc"
  >
    <span
      class="sc-HzRMQ jvqWRJ"
    />
  </div>
</DocumentFragment>
`;
